jcl/src/java.base/share/classes/java/lang/Thread.java

```diff
@@ -180,6 +186,9 @@ private Thread(String vmName, Object vmThreadGroup, int vmPriority, boolean vmIs
  	
 	/*[PR 100718] Initialize System.in after the main thread */
 	if (booting) {
+		/*[IF JAVA_SPEC_VERSION >= 15]*/
+		setNameImpl(threadRef, threadName);
```

Comment:
Sorry, one more think. Please add a comment to explain why the Java 15 code is different.

Reply:
Added a comment to explain the rational of this change.